src/golang/lib/execution_environment/integration.go

```diff
@@ -0,0 +1,296 @@
+	integrationWriter integration.Writer,
+	db database.Database,
+) {
+	now := time.Now()
+	_, err := integrationWriter.UpdateIntegration(
+		ctx,
+		integrationID,
+		map[string]interface{}{
+			"config": (*collection_utils.Config)(&map[string]string{
+				ExecStateKey: serializedRunning(&now),
+			}),
+		},
+		db,
+	)
+	if err != nil {
+		log.Errorf("Failed to update conda integration: %v", err)
+		return
+	}
+
+	out, _, err := lib_utils.RunCmd(CondaCmdPrefix, "info", "--base")
+	if err != nil {
+		updateFailure(
+			ctx,
+			out,
+			err.Error(),
+			"", /* condaPath */
+			&now,
+			integrationID,
+			integrationWriter,
+			db,
+		)
+
+		return
+	}
+
+	condaPath := strings.TrimSpace(out)
+	log.Infof("Conda base path is %s", condaPath)
+
+	pythonVersions := []string{"3.7", "3.8", "3.9", "3.10"}
+	for _, pythonVersion := range pythonVersions {
+		args := []string{
+			"create",
+			"-n",
+			fmt.Sprintf("aqueduct_python%s", pythonVersion),
+			fmt.Sprintf("python==%s", pythonVersion),
+			"-y",
+		}
+		_, _, err := lib_utils.RunCmd(CondaCmdPrefix, args...)
+		if err != nil {
+			updateFailure(
+				ctx,
+				out,
+				err.Error(),
+				condaPath,
+				&now,
+				integrationID,
+				integrationWriter,
+				db,
+			)
+
+			return
+		}
+
+		args = []string{
+			"run",
+			"-n",
+			fmt.Sprintf("aqueduct_python%s", pythonVersion),
+			"pip3",
+			"install",
+			fmt.Sprintf("aqueduct-ml==%s", lib.ServerVersionNumber),
+		}
+		_, _, err = lib_utils.RunCmd(CondaCmdPrefix, args...)
+		if err != nil {
+			updateFailure(
+				ctx,
+				out,
+				err.Error(),
+				condaPath,
+				&now,
+				integrationID,
+				integrationWriter,
+				db,
+			)
+
+			return
+		}
+	}
+
+	// This is to ensure we can use `conda develop` to update the python path later on.
+	args := []string{
+		"install",
+		"conda-build",
+		"-y",
+	}
+	_, _, err = lib_utils.RunCmd(CondaCmdPrefix, args...)
+	if err != nil {
+		updateFailure(
+			ctx,
+			out,
+			err.Error(),
+			condaPath,
+			&now,
+			integrationID,
+			integrationWriter,
+			db,
+		)
+
+		return
+	}
+
+	_, err = integrationWriter.UpdateIntegration(
+		ctx,
+		integrationID,
+		map[string]interface{}{
+			"config": (*collection_utils.Config)(&map[string]string{
+				CondaPathKey: condaPath,
+				ExecStateKey: serializedSuccess(&now),
+			}),
+		},
+		db,
+	)
+
+	if err != nil {
+		log.Errorf("Failed to update conda integration: ")
+	}
+}
+
+func getCondaIntegration(
+	ctx context.Context,
+	userId uuid.UUID,
+	integrationReader integration.Reader,
+	db database.Database,
+) (*integration.Integration, error) {
+	integrations, err := integrationReader.GetIntegrationsByServiceAndUser(
+		ctx,
+		integration.Conda,
+		userId,
+		db,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(integrations) == 0 {
+		return nil, nil
+	}
+
+	return &integrations[0], nil
+}
+
+// CondaConnectionState shows the current conda connection status
+// if the user is connected to conda. Otherwise, it returns nil.
+func CondaConnectionState(
```

Comment:
Since `integration.Integration` contains `shared.ExecutionState`, it seems cleaner (remove duplicate) to replace this function with a function like `extractExecState` that takes in `integration.Integration` and returns its exec state. The caller in `preview.go` can first call `getCondaIntegration` to get the integration object, and then pass it to `extractExecState` to get the exec state.